### DISCUSSION TITLE
Canvas Border Visibility, UI, Grids, magnetic Line snapping

### DISCUSSION THREAD

**Discussion by Rohitgit2412** [keiwando/toonsquid#621] [Ideas]
Writing this to express my appreciation for ToonSquid and its contribution to the animation industry. It's a fantastic app giving tough competetion to even professional industry standard animation softwares. Many many thanks to the genius Keiwan one man army behind this app.
I have some feature requests that I believe would significantly enhance the user experience:
 * Canvas Border Visibility: Pixel/raster images when used, particularly extensive backgrounds, it overlap the canvas border line, making it difficult to gauge its true perspective. I propose that the canvas border remains visible on top of every layer at all times.
 * UI: The adding of more drawing/track layers is the MOST frequently used option in animation/drawing app. So IMO it should not be two step process by first tapping on + icon then selecting the 'add layer' as it becomes annoying sometimes. This should be directly available outside on the sidebar or somewhere(for both adding drawing layer and track layer). And pen tool icon also which is frequently used should be somewhere up in the list because timeline panel hides the pen tool. It should be atleast available above the text icon.
 * Grids with Magnetic Snapping: The addition of grids above the white canvas with magnetic line snapping functionality (horizontal, vertical, and diagonal angles) would be incredibly useful for creating precise and aligned artwork.
 * Keyframe Effects: The ability to add keyframes for effects like blur, noise, HSB/warping and rigging would be icing on the cake.
I understand these may be ambitious requests, but they would undoubtedly make ToonSquid an even more powerful and versatile animation tool.
Thank you for your time and for making animation accessible and enjoyable.

**Comment by keiwando** [comment]
Thanks for the suggestions!
